### PR TITLE
HW-04_done

### PR DESCRIPTION
**Pull request by BukhtikYu**

### CODE REVIEW

File: HW04/Yuliya_Bukhtik/my_module_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  describe '#my_select' do
+    it 'iterates over the array and returnes new array with even numbers' do
+      expect(a.my_select).to eq [2, 4]
+    end
+  end
+
+  describe '#my_each' do
+    it 'returnes new array with square of each element' do
```
Comment (vladhilko) [inline HW04/Yuliya_Bukhtik/my_module_spec.rb:24]: Are you sure? 🤔 maybe `my_map`?

File: HW04/Yuliya_Bukhtik/my_module_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+require 'spec_helper'
+
+require_relative 'array.rb'
+require_relative 'my_module.rb'
+
+RSpec.describe MyModule do
+  subject(:a) { Array.new([1, 2, 3, 4]) }
+
+  describe '#my_map' do
+    it 'iterates over the array and prints each element' do
```
Comment (vladhilko) [inline HW04/Yuliya_Bukhtik/my_module_spec.rb:12]: Maybe `my_each`?

File: HW04/Yuliya_Bukhtik/my_module_spec.rb (hunk 1 above)
Comment (BukhtikYu) [reply HW04/Yuliya_Bukhtik/my_module_spec.rb:24]: yes, you are right.  I changed #my_map on #my_each.                                                         
#each doesn't return any result,but my #map does.

File: HW04/Yuliya_Bukhtik/my_module.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module MyModule
+  def my_each
+    each do |item|
```
Comment (vladhilko) [inline HW04/Yuliya_Bukhtik/my_module.rb:5]: https://github.com/labs-ruby/homeworks-2020/tree/master/HW04
Looks like we can't use `each`
`Important: Do not use #each, #select, #map in your implementation.`

Comment (vladhilko) [reply HW04/Yuliya_Bukhtik/my_module.rb:5]: And try to use blocks in your implementation

**Pull request merged**